app/src/main/java/org/thoughtcrime/securesms/onboarding/recoverypassword/RecoveryPasswordActivity.kt

```diff
@@ -0,0 +1,218 @@
+    }
+
+    private fun onHide() {
+        showSessionDialog {
+            title(R.string.recoveryPasswordHidePermanently)
+            htmlText(R.string.recoveryPasswordHidePermanentlyDescription1)
+            destructiveButton(R.string.continue_2, R.string.AccessibilityId_continue) { onHideConfirm() }
+            cancelButton()
+        }
+    }
+
+    private fun onHideConfirm() {
+        showSessionDialog {
+            title(R.string.recoveryPasswordHidePermanently)
+            text(R.string.recoveryPasswordHidePermanentlyDescription2)
+            cancelButton()
+            destructiveButton(
+                R.string.yes,
+                contentDescription = R.string.AccessibilityId_confirm_button
+            ) {
+                viewModel.permanentlyHidePassword()
+                finish()
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+fun PreviewRecoveryPassword(
+    @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+) {
+    PreviewTheme(themeResId) {
+        RecoveryPassword(seed = "Voyage  urban  toyed  maverick peculiar tuxedo penguin tree grass building listen speak withdraw terminal plane")
+    }
+}
+
+@Composable
+fun RecoveryPassword(
+    seed: String = "",
+    copySeed:() -> Unit = {},
+    onHide:() -> Unit = {}
+) {
+    AppTheme {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            modifier = Modifier
+                .contentDescription(R.string.AccessibilityId_recovery_password)
+                .verticalScroll(rememberScrollState())
+                .padding(bottom = 16.dp)
+        ) {
+            SmallButtons {
+                RecoveryPasswordCell(seed, copySeed)
+                HideRecoveryPasswordCell(onHide)
+            }
+        }
+    }
+}
+
+@Composable
+fun RecoveryPasswordCell(seed: String, copySeed:() -> Unit = {}) {
+    var showQr by remember {
+        mutableStateOf(false)
+    }
+
+    CellWithPaddingAndMargin {
+        Column {
+            Row {
+                Text(stringResource(R.string.sessionRecoveryPassword))
+                Spacer(Modifier.width(8.dp))
+                SessionShieldIcon()
+            }
+
+            Text(stringResource(R.string.recoveryPasswordDescription))
+
+            AnimatedVisibility(!showQr) {
+                Text(
+                    seed,
+                    modifier = Modifier
+                        .contentDescription(R.string.AccessibilityId_recovery_password_container)
+                        .padding(vertical = 24.dp)
+                        .border(
+                            width = 1.dp,
+                            color = classicDarkColors[3],
+                            shape = RoundedCornerShape(11.dp)
+                        )
+                        .padding(24.dp),
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.small.copy(fontFamily = FontFamily.Monospace),
```

Comment:
This is the wrong style. It is set to extraSmall in figma.
Also, are we reusing this type of text elsewhere? If so it could be made more generic, as a styling or as a composable.